rfcs/2021-07-20-8288-csv-enrichment.md

```diff
@@ -0,0 +1,202 @@
+# RFC 8288 - 2021-07-20 - CSV Enrichment
+
+Vector needs to allow users to enrich the events flowing through the topology
+using a CSV file. This RFC proposes adding a VRL function that can perform 
+this lookup.
+
+## Scope
+
+The RFC will cover a simple VRL function that will lookup a single row from a
+CSV file. 
+
+The lookup will be a simple sequential scan. The first version of this
+function will not handle any indexing of this data. Above a certain size
+indexing will provide substantial performance benefits, so it is something that
+we should definitely plan for implementing at a future stage. The RFC will 
+briefly discuss this, but will not go into any specific indexing strategies.
+
+## Motivation
+
+Users would like to enrich events flowing through Vector with extra data 
+provided by a CSV file. 
+
+## Doc-level Proposal
+
+### Resources
+
+To represent the CSV file we have a new top level configuration option.
+
+```
+[resources.csv_file]
+  type = "csv"
+  path = "\path_to_csv"
+  key = "{KEY}"
+  delimeter = ","
+```
+
+The fields available for this section are:
+
+*type* 
+
+The type of data that the resource represents. Currently only a csv file is 
+supported, but this may expand in future to handle other types of files and
+database connections.
+
+*path*
+
+The path to the csv file, either an absolute path or one relative to the current
+working directory.
+
+*key*
+
+The CSV file may be encrypted. This parameter specifies a key to use to decrypt
+the file.
+
+Is it safe to assume/require the file to be encrypted using AES, or should we
+support other algorithms as well?
+
+*delimeter*
+
+The delimeter used in the csv file to separate fields. Defaults to `","`.
+
+*title_row*
+
+Defaults to true. If set it assumes the first row in the csv file contains 
+column names. If false columns are named according to their numerical index.
+
+### Vrl functions
+
+Two remap functions:
+
+#### `lookup_csv_row`
```

Comment:
The `lookup_*` prefix is a new convention and I don't think it's the best term. How about the following prefixes?

* `find_*` - finds a _single_ record
* `search_*` - finds _multiple_ records

This will set precedence for future functions that read from other sources, like a database.